test/integration/http_integration.cc

```diff
@@ -978,11 +1044,13 @@ void HttpIntegrationTest::testTwoRequests(bool network_backup) {
   // created while the socket appears to be in the high watermark state, and regression tests that
   // flow control will be corrected as the socket "becomes unblocked"
   if (network_backup) {
-    config_helper_.addFilter(R"EOF(
-  name: pause-filter
+    config_helper_.addFilter(
+        fmt::format(R"EOF(
+  name: pause-filter{}
   typed_config:
     "@type": type.googleapis.com/google.protobuf.Empty
-  )EOF");
+  )EOF",
+                    downstreamProtocol() == Http::CodecClient::Type::HTTP3 ? "-for-quic" : ""));
```

Comment:
preference for quic logic in the normal pause filter
I used the connection ... protocol() as a detect option for mine, may work for you?

Reply:
pause filter library is depended by http_integration_lib, we want them to be able to compile out QUICHE right?

Reply:
ah yeah, so merging it in will have to wait until the compile macros land, good point.
do we really need all the custom down-casting rather than just forcing callbacks?  I had missed the degree of quic-specific logic here.